src/lib.rs

```diff
@@ -17,8 +20,94 @@ use tempfile::TempDir;
 use tracing::{debug, info, instrument, warn};
 use url::Url;
 
+use hc_utils::WrappedHeaderHash;
+use holochain::conductor::api::AppResponse;
+use holochain::conductor::api::ZomeCall;
+use holochain_zome_types::{
+    zome::{FunctionName, ZomeName},
+    zome_io::ExternInput,
+    SerializedBytes,
+};
 type HappIds = Vec<String>;
 
+pub async fn activate_holo_hosted_happs(core_happ: Happ) -> Result<()> {
+    let list_of_happs = get_enabled_hosted_happs(core_happ).await?;
+    install_holo_hosted_happs(list_of_happs).await?;
+    Ok(())
+}
+
+pub async fn install_holo_hosted_happs(
+    happs: impl Iterator<Item = WrappedHeaderHash>,
+) -> Result<()> {
+    info!("Starting to install....");
+    // iterate through the vec and
+    // Call http://localhost/hpos-holochain-api/install_hosted_happ
+    // for each WrappedHeaderHash to install the hosted_happ
+    let client = reqwest::Client::new();
+    // Note: Tmp preferences
+    let preferences = Preferences {
+        max_fuel_before_invoice: 1.0,
+        max_time_before_invoice: vec![86400, 0],
+        price_compute: 1.0,
+        price_storage: 1.0,
+        price_bandwidth: 1.0,
+    };
+    for happ_id in happs {
+        info!("Installing happ-id {:?}", happ_id);
+        let body = Body {
+            happ_id: happ_id.0.to_string(),
+            preferences: preferences.clone(),
+        };
+        let response = client
+            .post("http://localhost/hpos-holochain-api/install_hosted_happ")
+            .json(&body)
+            .send()
+            .await?;
+        info!("Installed happ-id {:?}", happ_id);
+        info!("Response {:?}", response);
+    }
+    Ok(())
+}
+
+pub async fn get_enabled_hosted_happs(
+    core_happ: Happ,
+) -> Result<impl Iterator<Item = WrappedHeaderHash>> {
+    let mut app_websocket = AppWebsocket::connect(42233)
+        .await
+        .context("failed to connect to holochain's app interface")?;
+    match app_websocket.get_app_info(core_happ.id_from_config()).await {
+        Some(holochain_types::prelude::InstalledApp {
+            // This works on the assumption that the core happs has HHA in the first position of the vec
+            cell_data,
+            ..
+        }) => {
+            let zome_call_payload = ZomeCall {
+                cell_id: cell_data[0].as_id().clone(),
+                zome_name: ZomeName::from("hha"),
+                fn_name: FunctionName::from("get_happs"),
+                payload: ExternInput::new(SerializedBytes::default()),
+                cap: None,
+                provenance: cell_data[0].clone().into_id().into_dna_and_agent().1,
+            };
```

Comment:
Let me try again at improving this cloning.
```suggestion
        Some(holochain_types::prelude::InstalledApp {
            cell_data,
            ..
        }) => {
            let hha_cell = cell_data.into_iter().next().unwrap_or_else(|| anyhow!("expected HHA to have at least one cell")?;
            let provenance = hha_cell.as_id().agent_pubkey().clone();
            let zome_call_payload = ZomeCall {
                cell_id: hha_cell.into_id(),
                zome_name: ZomeName::from("hha"),
                fn_name: FunctionName::from("get_happs"),
                payload: ExternInput::new(SerializedBytes::default()),
                cap: None,
                provenance,
            };
```